code/datums/ai/monkey/monkey_controller.dm

```diff
@@ -0,0 +1,211 @@
+		current_movement_target = W
+		current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/monkey_equip/ground)
+		return TRUE
+	else
+		var/mob/living/carbon/human/H = locate(/mob/living/carbon/human/) in oview(2,living_pawn)
+		if(H)
+			W = pick(H.held_items)
+			if(W && !blackboard[BB_MONKEY_BLACKLISTITEMS][W] && W.force > blackboard[BB_MONKEY_BEST_FORCE_FOUND])
+				blackboard[BB_MONKEY_PICKUPTARGET] = W
+				current_movement_target = W
+				current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/monkey_equip/pickpocket)
+				return TRUE
+
+//When idle just kinda fuck around.
+/datum/ai_controller/monkey/PerformIdleBehavior(delta_time)
+	var/mob/living/living_pawn = pawn
+
+	if(DT_PROB(25, delta_time) && (living_pawn.mobility_flags & MOBILITY_MOVE) && isturf(living_pawn.loc) && !living_pawn.pulledby)
+		step(living_pawn, pick(GLOB.cardinals))
+	else if(DT_PROB(5, delta_time))
+		INVOKE_ASYNC(living_pawn, /mob.proc/emote, pick("screech"))
+	else if(DT_PROB(1, delta_time))
+		INVOKE_ASYNC(living_pawn, /mob.proc/emote, pick("scratch","jump","roll","tail"))
+
+///Reactive events to being hit
+/datum/ai_controller/monkey/proc/retaliate(mob/living/L)
+	var/list/enemies = blackboard[BB_MONKEY_ENEMIES]
+	enemies[L] += MONKEY_HATRED_AMOUNT
+
+/datum/ai_controller/monkey/proc/on_attackby(datum/source, obj/item/I, mob/user)
+	SIGNAL_HANDLER
+	if(I.force && I.damtype != STAMINA)
+		retaliate(user)
+
+/datum/ai_controller/monkey/proc/on_attack_hand(datum/source, mob/living/L)
+	SIGNAL_HANDLER
+	if(L.a_intent == INTENT_HARM && prob(MONKEY_RETALIATE_HARM_PROB))
+		retaliate(L)
+	else if(L.a_intent == INTENT_DISARM && prob(MONKEY_RETALIATE_DISARM_PROB))
+		retaliate(L)
+
+/datum/ai_controller/monkey/proc/on_attack_paw(datum/source, mob/living/L)
+	SIGNAL_HANDLER
+	if(L.a_intent == INTENT_HARM && prob(MONKEY_RETALIATE_HARM_PROB))
+		retaliate(L)
+	else if(L.a_intent == INTENT_DISARM && prob(MONKEY_RETALIATE_DISARM_PROB))
+		retaliate(L)
+
+/datum/ai_controller/monkey/proc/on_bullet_act(datum/source, obj/item/projectile/Proj)
+	SIGNAL_HANDLER
+	var/mob/living/living_pawn = pawn
+	if(istype(Proj , /obj/item/projectile/beam)||istype(Proj, /obj/item/projectile/bullet))
+		if((Proj.damage_type == BURN) || (Proj.damage_type == BRUTE))
+			if(!Proj.nodamage && Proj.damage < living_pawn.health && isliving(Proj.firer))
+				retaliate(Proj.firer)
+
+/datum/ai_controller/monkey/proc/on_hitby(datum/source, atom/movable/AM, skipcatch = FALSE, hitpush = TRUE, blocked = FALSE, datum/thrownthing/throwingdatum)
+	SIGNAL_HANDLER
+	if(istype(AM, /obj/item))
+		var/mob/living/living_pawn = pawn
+		var/obj/item/I = AM
+		if(I.throwforce < living_pawn.health && ishuman(I.thrownby))
+			var/mob/living/carbon/human/H = I.thrownby
+			retaliate(H)
+
+/datum/ai_controller/monkey/proc/on_Crossed(datum/source, atom/movable/AM)
+	var/mob/living/living_pawn = pawn
+	if(!living_pawn.incapacitated() && ismob(AM))
+		var/mob/living/in_the_way_mob = AM
+		in_the_way_mob.knockOver(living_pawn)
```

Comment:
This is what causes the excessive tripping I saw in testing, since there's like a dozen monkeys in confined spaces. So I think you just need to not have monkeys trip over other monkeys.

```suggestion
	if(!living_pawn.incapacitated() && ismob(AM) && !ismonkey(AM))
		var/mob/living/in_the_way_mob = AM
		in_the_way_mob.knockOver(living_pawn)
```

Reply:
Remove it entirely or maybe just add a prob(10) or some shit so it happens rarely?

Reply:
Yeah I edited my code suggestion to be a prob.